### PR TITLE
fix(sql): fix sql error [col in (1,2)] when col is type short

### PR DESCRIPTION
**Pull request by ideoma**
Fixes `UnsupportedOperationException` when running 

```SQL
CREATE table abc (a int, b short) 

insert into abc values(1, 2)

select * from abc where a in (1, 2)

```

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InLongFunctionFactory.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InLongFunctionFactory.java|1|1|100.00%|

File: core/src/test/java/io/questdb/test/log/LogAlertSocketTest.java [hunk 1]
```diff
@@ -172,7 +172,7 @@ public void testFailOverSingleHost() throws Exception {
             final SOCountDownLatch haltLatch = new SOCountDownLatch(1);
             final CyclicBarrier startBarrier = new CyclicBarrier(2);
             final MockAlertTarget server = new MockAlertTarget(
-                    0,
+                    9863,
```
Comment (jerrinot) [inline core/src/test/java/io/questdb/test/log/LogAlertSocketTest.java:175]: just for my curiosity: is this meant to fix the test flakiness? how?

**Review by jerrinot — APPROVED** [review]

File: core/src/test/java/io/questdb/test/log/LogAlertSocketTest.java (hunk 1 above)
Comment (ideoma) [reply core/src/test/java/io/questdb/test/log/LogAlertSocketTest.java:175]: Yes, I ran it on my Mac and with 0 (random) I got a sort of port exhaustion at some point, so decided to pin the port

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/InLongFunctionFactory.java|1|1|100.00%|

**Pull request merged**